src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String[] data = readFile(fromFileName);
+            String report = createReport(data);
+            writeToFile(report, toFileName);
+            System.out.println("Statistics generated and written to " + toFileName);
+        } catch (IOException e) {
```

Comment:
It's better to handle exception in place - read and write file methods

Reply:
fixed, but still handling